### PR TITLE
Fix failure on Unix associate with converting PDBs

### PR DESCRIPTION
**Pull request by vancem**
Code to convert portable pdbs to windows pdbs was being run on UNIX machines.
This is clearly not needed and was causing failures.   Fix it to only run on non Windows
and only for coverage (which is the only thing we know that needs the Windows  PDBs).

Basically just make the conversion condition stricter.  

See also  https://github.com/dotnet/corefx/issues/24807.

### CODE REVIEW

**Comment by vancem** [comment]
This is to fix bug https://github.com/dotnet/corefx/issues/24807

@dagood

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/CodeCoverage.targets [hunk 1]
```diff
@@ -50,11 +50,14 @@
...
   </Target>
 
-  <Target Name="GenerateWindowsPdbsForAssemblyBeingTested" BeforeTargets="GenerateTestExecutionScripts">
+  <!-- *********************************************************************************************** -->
     <!-- As of 10/2017 OpenCover does not support portable PDBs, but we want the builds to generate
          portable PDBs.  Thus we generate windows PDBs from portable PDBs here.   Can be removed
          when OpenCover directly supports Portable PDBs (probably by early 2018) -->
 
+  <Target Name="GenerateWindowsPdbsForAssemblyBeingTested" BeforeTargets="GenerateTestExecutionScripts"
+    Condition="$(CodeCoverageEnabled) == 'true' and '$(TargetOS)'=='Windows_NT'">
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/CodeCoverage.targets]: Not that we have a style guide, but I've usually seen chopping done like this in MSBuild:

```xml
<Target Name="GenerateWindowsPdbsForAssemblyBeingTested"
        BeforeTargets="GenerateTestExecutionScripts"
        Condition="'$(CodeCoverageEnabled)' == 'true' and '$(TargetOS)'=='Windows_NT'">
```

Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/CodeCoverage.targets]: `$(CodeCoverageEnabled)` is missing quotes in the condition.

**Comment by vancem** [comment]
@dotnet-bot test this please

**Pull request merged**